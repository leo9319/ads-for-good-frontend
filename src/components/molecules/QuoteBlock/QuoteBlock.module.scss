@import '../../../styles/breakpoints';

.quoteBlockContainer {
  padding: 0 var(--spacing-2xl);
  box-sizing: border-box;
  border-left: 2px solid var(--borderColor);
  color: var(--textColor);
}

@media (max-width: $xs-min) {
  .quoteBlockContainer {
    padding: 0 var(--spacing-lg);
  }
}
