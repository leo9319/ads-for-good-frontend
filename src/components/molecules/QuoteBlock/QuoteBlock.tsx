import React from 'react';
import Text from '@radix-styles/atoms/Text';
import { classNames, getModuleClasses } from '@utils/common/classNames';
import Box from '@radix-styles/atoms/Box';

import { getStylesAsCssProperties } from '@utils/common/styles';
import { TextSize } from '@internal/types/common/style';
import styles from './QuoteBlock.module.scss';

export interface QuoteBlockProps {
  /**
   * Quote text for the Quote Block
   * @property
   */
  quoteText: string;
  /**
   * Quote text color for the Quote Block
   * @property
   */
  quoteTextColor?: string | undefined;
  /**
   * Quote text size for the Quote Block
   * @property
   */
  quoteTextSize?: TextSize;
  /**
   * Border Color for the Quote Block
   * @property
   */
  borderColor?: string | undefined;
  /**
   * Class name for Quote Block container
   * @property
   */
  className?: string;
}

/**
 * @group Molecules
 * @category Component
 *
 * Quote Block component is used to display a Quote Block in Article pages
 * @param {QuoteBlockProps} props
 * @returns {ReactElement}
 */

export const QuoteBlock = ({
  quoteText,
  quoteTextColor = '#000',
  quoteTextSize = '4',
  borderColor = '#E86100',
  className,
}: QuoteBlockProps) => {
  const classes = getModuleClasses(className?.trim(), styles);
  return (
    <Box
      className={classNames(styles.quoteBlockContainer, classes)}
      style={getStylesAsCssProperties({
        '--borderColor': borderColor,
        '--textColor': quoteTextColor,
      })}
    >
      <Text size={quoteTextSize}>{quoteText}</Text>
    </Box>
  );
};

export default QuoteBlock;
