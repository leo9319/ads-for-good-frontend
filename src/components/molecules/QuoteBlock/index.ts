export { default } from './QuoteBlock';
export { QuoteBlock } from './QuoteBlock';
