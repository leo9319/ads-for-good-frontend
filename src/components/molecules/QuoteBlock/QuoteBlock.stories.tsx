import React from 'react';
import QuoteBlock from './QuoteBlock';
import { Meta, StoryObj } from '@storybook/react/*';

const meta = {
  title: 'Library/Molecules/QuoteBlock',
  component: QuoteBlock,
  tags: ['autodocs'],
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    quoteText: {
      control: 'text',
    },
    quoteTextColor: {
      control: 'color',
    },
    quoteTextSize: {
      control: {
        type: 'select',
      },
      options: [
        '1',
        '2',
        '3',
        '4',
        '5',
        '6',
        '7',
        '8',
        '9',
        '10',
        '11',
        '12',
        '13',
      ],
    },
    borderColor: {
      control: 'color',
    },
    className: {
      control: 'text',
    },
  },
  decorators: [
    Story => (
      <div>
        <Story />
      </div>
    ),
  ],
} satisfies Meta<typeof QuoteBlock>;

export default meta;

type Story = StoryObj<typeof meta>;

export const QuoteBlockContent: Story = {
  args: {
    quoteText:
      'We stand by the belief that if we have the ability to help, we have the responsibility to act. This ethos shapes our organization’s approach to humanitarian work, emphasizing not just charity, but sustainable development, empowerment, and advocacy. ',
    quoteTextColor: '#000',
    quoteTextSize: '4',
    borderColor: '#E86100',
    className: '',
  },
};
