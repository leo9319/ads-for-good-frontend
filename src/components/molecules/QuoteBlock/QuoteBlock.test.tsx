import React from 'react';
import QuoteBlock from './QuoteBlock';
import { render } from '@testing-library/react';

describe('QuoteBlock', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<QuoteBlock quoteText="Quote Block text" />);
    expect(baseElement).toMatchSnapshot();
  });
});
