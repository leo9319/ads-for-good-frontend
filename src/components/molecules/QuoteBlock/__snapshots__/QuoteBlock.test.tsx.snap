// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteBlock should render successfully 1`] = `
<body>
  <div>
    <div
      class="rt-Box quoteBlockContainer"
      style="--borderColor: #E86100; --textColor: #000;"
    >
      <span
        class="rt-Text mocked-class-name typography-font-family-default"
        id=""
      >
        Quote Block text
      </span>
    </div>
  </div>
</body>
`;
